incubator/java-spring-boot2-rhr/image/project/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM {{APPSODY_STACK}} as compile
+
+#setup project folder for java build step
+COPY . /project
+
+WORKDIR /project/user-app
+
+USER root
+
+RUN /project/util/check_version build \
+ && /project/java-spring-boot2-build.sh package
+
+USER java_user
+
+####
+FROM registry.redhat.io/redhat-openjdk-18/openjdk18-openshift
```

Comment:
> the final image requires Red Hat creds to pull

This is the image in question?  If this requires credentials/auth to pull then I don't think it belongs in `github.com/kabanero-io` does it? As an open source project we expect every Collection here to be usable by the general open source community.

Or have I missed something?

Reply:
@ianpartridge  No you have not missed anything, it was highlighted in the PR description. 
However this will require any user attempting to build this stack to have a redhat login. 
This is definitely not something we want to enforce and hence why we went with the UBI images.

Reply:
The `kabanero` Docker images that are produced by the `ci/build.sh` process and pushed to Docker Hub will still be based on UBI won't they?  It's only the final application built by the user via `appsody build` that will require the credentials.

How will the credentials be passed through from `appsody build`? Is the Red Hat registry accessible via a standard `docker login` or is there some different auth process?

Reply:
Standard docker login works with Red Hat account credentials. 

`docker login registry.redhat.io`

If the user has done this before invoking `appsody build` then it functions as expected. (Build engines etc would need appropriate creds if they wish to validate that capability, which is exactly why this PR fails in travis today). 

I've been told that the use of the RHR image as a base for the final image is required for the final image to be considered as supported. (This combined with moving the Spring dependencies from the Spring defined ones to the Red Hat ones defined via Snowdrop). Being considered as supported is the primary reason for this change, and partly is why changes are being made in a new stack, rather than as modifications to the existing `java-spring-boot2` stack.

Reply:
Thanks, that's great. Are we planning to give the Travis CI here some valid credentials so the CI can run?

Reply:
Good question, but not one I can answer in the scope of this PR =) 

Having travis able to do that makes sense, (especially if other RHR based stacks are due to follow) but extreme care would have to be taken that the creds are only used for the final `appsody build` test, and never made available during the stack build phase, else there's a risk that a stack could base itself on the RHR authenticated image, leading to unrestricted distribution of such a derived image. 

It may be better to discuss adding such a capability via a separate Issue, as it affects the build process generically rather than uniquely for this PR.

Reply:
Sounds good. Thanks for discussing all this. For now I think we leave this PR in draft?